lmp/tokenizer/_base_list_tokenizer.py

```diff
@@ -356,7 +416,27 @@ def build_vocab(
             min_count:
                 Minimum of token's frequence. If token's frequence is smaller
                 than `min_count`, then discard that token.
+
+        Raises:
+            TypeError:
+                When `batch_sequences` is not instance of `Iterable[str]` or
+                `min_count` is not instance of `int`.
         """
+        # Type check.
+        if not isinstance(batch_sequences, Iterable):
+            raise TypeError(
+                '`batch_sequences` must be instance of `Iterable[str]`.')
+
+        batch_sequences = list(batch_sequences)
+
+        if any([not isinstance(sequence, str) for sequence in batch_sequences]):
```

Comment:
Change `any([not isinstance...])` to `not all([isinstance ...])` improve performance by exploiting short connection.
Do the same for rest of the code.